core/src/test/java/se/tre/freki/labels/LabelClientTypeContextTest.java

```diff
@@ -152,4 +155,67 @@ public void createIdPublishesEventOnSuccess() throws Exception {
     typeContext.createId("foo").get();
     verify(eventBus).post(any(LabelCreatedEvent.class));
   }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testRenameNewNameExists() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+    // Can not rename to a name that is already taken
+    typeContext.createId("newName");
+    typeContext.rename("oldName", "newName");
+  }
+
+  @Test
+  public void testRenameIdNotFoundOnOldName() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+    //No previous label called "oldName" return Failed future
+    ListenableFuture<Void> response = typeContext.rename("oldName", "newName");
+
+    Futures.addCallback(response, new FutureCallback<Void>() {
+      @Override
+      public void onSuccess(final Void result) {
+        fail();
+      }
+
+      @Override
+      public void onFailure(final Throwable throwable) {
+        // Do nothing nwe should fail.
+      }
+    });
+  }
+
+  @Test
+  public void testRename() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+
+    final LabelId labelId = typeContext.createId("oldName").get();
+    ListenableFuture<Void> response = typeContext.rename("oldName", "newName");
+
+    Futures.addCallback(response, new FutureCallback<Void>() {
```

Comment:
See above about not blocking the calls for futures. In any case it is probably less verbose to just block and work with the result directly. Also, the contract for `Store#renameLabel` is that if one checks that the current name exists and the new name does not exist before a call to it, then if the `#renameLabel` call returns successfully and the future does not contain an exception, then the label should have been renamed. This means it is not strictly necessary to test that the label is renamed here, it's enough to call `#rename` and block on it to make sure it does not contain an exception.